src/compiler/types.ts

```diff
@@ -1485,11 +1485,11 @@ module ts {
     // Class and interface types (TypeFlags.Class and TypeFlags.Interface)
     export interface InterfaceType extends ObjectType {
         typeParameters: TypeParameter[];           // Type parameters (undefined if non-generic)
-        declaredProperties: Symbol[];              // Declared members
-        declaredCallSignatures: Signature[];       // Declared call signatures
-        declaredConstructSignatures: Signature[];  // Declared construct signatures
-        declaredStringIndexType: Type;             // Declared string index type
-        declaredNumberIndexType: Type;             // Declared numeric index type
+        declaredProperties: Symbol[];              // Declared members (computed by resolveDeclaredMembers)
+        declaredCallSignatures: Signature[];       // Declared call signatures (computed by resolveDeclaredMembers)
+        declaredConstructSignatures: Signature[];  // Declared construct signatures (computed by resolveDeclaredMembers)
+        declaredStringIndexType: Type;             // Declared string index type (computed by resolveDeclaredMembers)
+        declaredNumberIndexType: Type;             // Declared numeric index type (computed by resolveDeclaredMembers)

```

Comment:
I think it is better to put it in a different type like ResolvedType. InterfaceTypeWithBaseTypes does the same thing. If you like, you can combine this and baseTypes into one type.